### PR TITLE
add sort by number of moves

### PR DESCRIPTION
**Pull request by gcentauri**
addresses issue #336 

the logic for the sorting menu could probably be refactored to be a bit cleaner, but adding the 'moves' check at the bottom seems to work.  happy to continue working on it if there's more to improve here!

### CODE REVIEW

**Comment by gcentauri** [comment]
i'd be happy to try writing some tests around this module as well, but i'm unfamiliar with the test framework.  i'll do a little bit of digging

File: src/components/drawers/GameChooserDrawer.js [hunk 1]
```diff
@@ -297,6 +298,10 @@ class GameChooserDrawer extends Component {
                             s = x1 < x2 ? -1 : +(x1 !== x2)
                         }
 
+                        if (property === 'moves') {
```
Comment (yishn) [inline src/components/drawers/GameChooserDrawer.js]: We can include this `if` clause as an `else if` between line 296 and 297.

**Comment by yishn** [comment]
Thank you for your contribution! It looks good, I just left some minor comments. 

As for testing, currently we have no framework in place for testing components. One solution would be to extract the logic for game sorting into a separate js file in `src/modules` and write tests for it. But if this is too much work, we can forgo it.

**Comment by yishn** [comment]
That's great, thank you very much!

> Some of the test game data available didn't seem to work with the SGF property scheme the selector is using.

What do you mean by that exactly?

> I'm wondering if it wouldn't be worth creating some kind of GameCollection class to hold games and implement the sort method on it? Maybe that is something you already have in mind.

Sabaki uses plain objects and arrays to encode SGF data, and I'd like to keep it like that.

**Comment by gcentauri** [comment]
> What do you mean by that exactly?

Well for example, the file euc-kr.gib created a game tree where the value of the BR property was 1±Þ.  Perhaps this has some meaning I don't understand, but it looked like the file indicated that the "Level" of the player was "17".  Regardless, that string wouldn't sort properly with the function there currently.  Perhaps I am missing a step in parsing files that aren't SGF?

> Sabaki uses plain objects and arrays to encode SGF data, and I'd like to keep it like that.

After thinking about it more I agree.  The important part is just the ability to compare, there's not really any need to abstract something else.

**Comment by fohristiwhirl** [comment]
> Well for example, the file euc-kr.gib created a game tree where the value of the BR property was 1±Þ

That's a failure to detect the proper encoding in the GIB. But in any case, properties like BR can in reality be completely arbitrary.

**Comment by yishn** [comment]
Maybe let's only use the SGF files for testing for now. I'd say the other formats don't have priority.

**Comment by gcentauri** [comment]
Okay, if you'd like some more robust tests, I can put together a small collection of SGF files to include in the repo.  As far as I know, game records are free to use data, correct?  Barring any commentary I suppose.  I have a collection of my own from KGS, and a large collection of Pro games.  Let me know if there's any preferred source for the actual data files.

**Comment by yishn** [comment]
That's a good idea, we probably just need a small subset of games for testing.

**Comment by gcentauri** [comment]
I created a few sgf files with Sabaki and changed the tests to sort them.  Hope it all looks okay!  Thanks for making Sabaki!

**Comment by yishn** [comment]
Thank you very much! I probably thought there would be no need to sort them with natsort. Natsort ensures that numbers are sorted correctly, e.g. that 'Game 10' comes after 'Game 2'. Since it's unlikely that player names contains numbers, I didn't sort them with natsort. But for consistency's sake, maybe we should... What do you think?

**Comment by gcentauri** [comment]
User names from go servers could certainly contain numbers, though the sequence is likely meaningless...  Shodan, Shodan1, Shodan01, Shodan2, Shodan10... I guess it isn't probably something a user would notice too much?

**Comment by gcentauri** [comment]
I am doing some more refactoring on the sort module, but I missed the trick for stable sort.  I will try to finish that tonight and push up another commit.  I went ahead and grouped all Name type properties to use natsort

**Comment by gcentauri** [comment]
Hope I didn't go too overboard refactoring... let me know what you think! I can squash all of this into a single commit if you prefer to merge that way.

**Comment by yishn** [comment]
I like the new API of `gamesort`, thank you again for your hard work!

**Pull request merged**